### DISCUSSION TITLE
Xonsh automatically exits shell on builtin command

### DISCUSSION THREAD

**Discussion by Serrindipity** [xonsh/xonsh#5688] [Q&A] [answered]
Whenever I run a xonsh builtin (i.e. xonfig -h, xontrib -h, etc.), the shell exits. I'm pretty new to all of this, so I'm not sure if I'm doing something wrong. This persists even when running xonsh --no-rc --no-env

Xonsh is installed with brew, version 0.18.3 on MacOS.

I've tried uninstalling and re-installing as well. Not sure what's happening.

**Comment by anki-code** [comment, marked as answer]
Hey! Nice catch! It's an issue. Moved to issue tracker - https://github.com/xonsh/xonsh/issues/5689
